libraries/rush-lib/src/logic/Autoinstaller.ts

```diff
@@ -233,4 +247,100 @@ export class Autoinstaller {
       console.log(message);
     }
   }
+
+  public static getAutoinstallerNames(rushConfiguration: RushConfiguration): string[] {
+    const autoinstallerNames: string[] = [];
+    const files: FolderItem[] = FileSystem.exists(rushConfiguration.commonAutoinstallersFolder)
+      ? FileSystem.readFolderItems(rushConfiguration.commonAutoinstallersFolder)
+      : [];
+
+    for (const file of files) {
+      if (
+        file.isDirectory() &&
+        FileSystem.exists(path.join(rushConfiguration.commonAutoinstallersFolder, file.name, 'package.json'))
+      ) {
+        autoinstallerNames.push(file.name);
+      }
+    }
+
+    return autoinstallerNames;
+  }
+
+  public static async prepareAutoinstallersAsync(
+    rushConfiguration: RushConfiguration,
+    specificNames?: string[]
+  ): Promise<void> {
+    const autoinstallerNames: string[] = [];
+
+    if (specificNames) {
+      autoinstallerNames.push(...specificNames);
+    } else {
+      autoinstallerNames.push(...Autoinstaller.getAutoinstallerNames(rushConfiguration));
+    }
+
+    for (const autoinstallerName of autoinstallerNames) {
+      const autoinstaller: Autoinstaller = new Autoinstaller({
+        autoinstallerName,
+        rushConfiguration
+      });
+      await autoinstaller.prepareAsync();
+      console.log('');
+    }
+
+    console.log(colors.gray('='.repeat(79)));
+
+    if (autoinstallerNames.length === 0) {
+      console.log('\nNo autoinstallers to prepare.');
+      return;
+    }
+
+    console.log(colors.green(`\n${autoinstallerNames.length} autoinstaller(s) prepared.`));
+  }
+
+  public static updateAutoinstallers(rushConfiguration: RushConfiguration, specificNames?: string[]): void {
+    const autoinstallerNames: string[] = [];
+
+    if (specificNames) {
+      autoinstallerNames.push(...specificNames);
+    } else {
+      autoinstallerNames.push(...Autoinstaller.getAutoinstallerNames(rushConfiguration));
+    }
+
+    const updated: Autoinstaller[] = [];
+    const skipped: Autoinstaller[] = [];
+
+    for (const autoinstallerName of autoinstallerNames) {
+      const autoinstaller: Autoinstaller = new Autoinstaller({
+        autoinstallerName,
+        rushConfiguration
+      });
```

Comment:
Will this error if the named autoinstaller doesn't exist?

Reply:
Yes -- it treats it essentially like a series of steps, for example:

```console
rush update-autoinstaller --name good1 --name good2 --name oops
```

Assuming `oops` is a misspelled autoinstaller name, the command would update good1, good2, and then dump out with the typical error. The user could then re-run the command (fixing the mistake), and it would end up printing: 2 autoinstallers skipped, 1 autoinstaller updated (since good1 and good2 had just been updated by the last run).

It would probably be a smidge nicer of an experience if we completely wrapped up (skips, successes, failures) in a bow and presented them, but failing install on an autoinstaller is unusual enough that it seemed unnecessary for this PR.